cargo-dist/src/platform.rs

```diff
@@ -79,6 +81,28 @@ pub enum RuntimeCondition {
     Rosetta2,
 }
 
+/// Runtime conditions specific to Linux
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+#[derive(Default)]
+pub struct LinuxRuntimeConditions {
+    /// The system glibc must be at least this version
+    pub glibc: Option<RuntimeCondition>,
+    /// The system musl libc must be at least this version
+    pub musl: Option<RuntimeCondition>,
+}
+
+
+/// Runtime conditions collected for every platform
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+#[derive(Default)]
+pub struct RuntimeConditions {
+    /// Linux-specific conditions
+    pub linux: LinuxRuntimeConditions,
+    /// MacOS-specific conditions
+    pub macos: Option<RuntimeCondition>,
+}
+
```

Comment:
per discussion: we should just drop the enum entirely in favour of a struct with many Option fields. It's also not clear to me that we need this level of platform distinction, since all the options apply to the platform uniformly

Reply:
Merged in your other PR that did this.